### PR TITLE
Change compression method for render dump

### PR DESCRIPTION
**Pull request by aTom3333**
Implement the FPC algorithm to compress the render dumps instead of using gzip.
Compression ratio is similar but with higher reading/writing speed (around 10x).
The render dump format is changed. Backward compatibility is kept by trying to read a magic number (to differentiate from an old dump) and if it doesn't match, puting back the bytes so they are present for the gzip stream

### CODE REVIEW

File: chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java [hunk 1]
```diff
@@ -89,6 +82,8 @@
  */
 public class Scene implements JsonSerializable, Refreshable {
 
+  private static final byte[] DUMP_FORMAT_MAGIC_NUMBER = {0x44, 0x55, 0x4D, 0x50};
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:85]: `DUMP` :smile:

File: chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      current = bits;
+
+      return Double.longBitsToDouble(bits);
+    }
+  }
+
+  public static void compress(double[] input, OutputStream output) throws IOException {
+    try(BufferedOutputStream out = new BufferedOutputStream(output)) {
+      if(input.length % 3 != 0)
+        throw new RuntimeException("Dump doesn't have a multiple of 3 values");
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java:138]: This should probably be an `IllegalArgumentException`

File: chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        gEncoder.encodeSingleWithOddTerminator(input[idx+1], out);
+        bEncoder.encodeSingleWithOddTerminator(input[idx+2], out);
+      }
+    }
+  }
+
+  public static void decompress(double[] output, InputStream input) throws IOException {
+    try(BufferedInputStream in = new BufferedInputStream(input)) {
+      if(output.length % 3 != 0)
+        throw new RuntimeException("Dump doesn't have a multiple of 3 values");
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java:164]: See above

File: chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      if(pixels % 2 == 1) {
+        int idx = 3*size;
+        rEncoder.encodeSingleWithOddTerminator(input[idx], out);
+        gEncoder.encodeSingleWithOddTerminator(input[idx+1], out);
+        bEncoder.encodeSingleWithOddTerminator(input[idx+2], out);
+      }
+    }
+  }
+
+  public static void decompress(double[] output, InputStream input) throws IOException {
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java:161]: (nit) For the `compress` method, it's `input, output`. Should the arguments be switched here?

**Review by leMaik — APPROVED** [review]

File: chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java (hunk 4 above)
Comment (aTom3333) [reply chunky/src/java/se/llbit/chunky/resources/FloatingPointCompressor.java:161]: Yeah, I let like that to get the types in the same order but I don't really care

**Pull request merged**